src/huggingface_hub/inference/_client.py

```diff
@@ -1309,3 +1318,52 @@ def _resolve_url(self, model: Optional[str] = None, task: Optional[str] = None)
             # Otherwise, we use the default endpoint
             else f"{INFERENCE_ENDPOINT}/models/{model}"
         )
+
+    def get_model_status(self, model: Optional[str] = None, *, token: Optional[str] = None) -> ModelStatus:
+        """
+            A function which returns the status os a specific model, from the huggingface Inference API.
+
+            Parameters:
+            -----------
+            model -> Optional[str], default=None
+                Identifier of the model for witch the status gonna be returned(retrieve). If model is not provided, 
+                the model associated with this instance of InferenceClient will be used. The 
+                identifier should not be a URL.
+
+            token -> Optional[str], default=None
+                 Optional token for authenticating with the Hugging Face API. If not provided, the 
+                 token associated with this instance of InferenceClient will be used.
+            
+                 
+            Raises
+            ------
+            ValueError
+                If the model is missing, meaning is not provided. Or the provided model is a URL. And finally
+                it raises an error if the API returns an error message(missing model).
+            
+                
+            Returns
+            -------
+            ModelStatus
+                An instance of ModelStatus dataclass, containing information about the state of 
+                the model: load, state, compute type and framework.
+        """
+        model = model or self.model
+        if model is None:
+             raise ValueError("Model id not provided")
+        if model.startswith("https://"):
+            raise ValueError("...")  # only works for InferenceAPI, not any URL 
+        
+        huggingface_interface_response = request.get(f"https://api-inference.huggingface.co/status/{model}")
```

Comment:
```suggestion
        huggingface_interface_response = get_session().get(f"{INFERENCE_ENDPOINT}/status/{model}")
```

Two things here:
- better to use `get_session` (will keep connection open between requests)
- you can use a constant value to build the URL

Reply:
Ok cool got you.